daemon/media_socket.c

```diff
@@ -418,13 +570,24 @@ static int get_port(socket_t *r, unsigned int port, struct intf_spec *spec) {
 		return ret;
 	}
 
+	pp->free_ports--;
```

Comment:
I think all operations on the free_ports var need to use atomic ops - volatile alone isn't enough. The g_atomic_int_* functions should be used, at least for inc and dec operations (but better to use them for everything)